### ISSUE TITLE
HDF5 AttributeError: 'ImageNet' object has no attribute 'init_ops'

### ISSUE THREAD

**Issue by anayebi** [neuroailab/tfutils#58]
When reading from the `imagenet2012.hdf5` file (such as when calling `train_alexnet.py` in tutorials), I get the error `'ImageNet' object has no attribute 'init_ops'`. 

The `HDF5DataReader` class in `data` should be updated to have an `init_ops()` attribute as is currently required for all other `DataProviderBase` objects.

**Comment by yamins81** [comment]
Actually, the HDF5Reader class is fine.  It's not meant to be a DataProvider.  Instead, the tutorial needs to be updated to use the ParallelBySliceDataProvider which will in turn use the HDF5Reader class.   See the usage pattern in the tests in test_data_hdf5.py.     @anayebi can you update the tests?

**Comment by qbilius** [comment]
I'll rewrite the tutorial to use tfrecords.

**Comment by anayebi** [comment]
Ok, I updated the `train_alexnet.py` tutorial to work with the `imagenet2012.hdf5` file, and I made the `ImageNet` class in `data.py` a subclass of `ParallelBySliceDataProvider` instead of `HDF5Reader`. I've committed this to the `alexnet_hdf5` branch. (Hesitant to PR at the moment, as I needed to make a hacky change to `base.py` to access specific groups in the hdf5 file depending on whether we are in training mode or in validation mode. I'll try and see if I can do without that and then PR.)
